src/orion/algo/tpe.py

```diff
@@ -283,7 +280,8 @@ def suggest(self, num=1):
                                                        above_points[idx: idx + shape[0]],
                                                        self._sample_categorical_point)
                 else:
-                    raise NotImplementedError()
+                    # fidelity dimension
```

Comment:
maybe `elif dimension.type == 'fidelity'` to be sure to catch unsupported dimensions that would have been missed in `__init__`